r2/r2/models/subreddit.py

```diff
@@ -540,6 +540,33 @@ def remove_moderator(self, user, **kwargs):
         hook = hooks.get_hook("subreddit.remove_moderator")
         hook.call(subreddit=self, user=user)
 
+        if user == self.top_moderator_account and len(self.moderators) > 1:
+            from r2.models import ModAction
+            fullperms = self.moderator_permission_class.loads('+all')
+            # cycle through each available moderator
+            for user_id in self.moderators[1:]:
+                successor = Account._byID(user_id)
+                # skip deleted / shadowbanned users
+                if successor._deleted or successor._spam:
+                    continue
+                successor_rel = self.get_moderator(successor)
+                successor_perms = successor_rel.get_permissions()
+                # if the successor does not have all perms, give it to him/her
+                if not successor_perms['all']:
```

Comment:
Just to make this a little simpler (and reduce indentation more), when you're doing this in the new function you can just do something more like:

    # if the successor already has full perms, we don't need to do anything
    if successor_perms["all"]:
        return

And then the code for setting permissions and creating the mod log entry doesn't need to be inside an `if:`.